apis/Google.Devtools.AspNet/Google.Devtools.AspNet.Tests/CloudErrorReporting/CloudErrorReportingExceptionLoggerTest.cs

```diff
@@ -0,0 +1,166 @@
+                e.Context.HttpRequest.Method.Equals(DeleteMethod.ToString()) &&
+                e.Context.HttpRequest.Url.Equals(GoogleUri.ToString()) &&
+                e.Context.HttpRequest.UserAgent.Equals(UserAgentValue.ToString()) &&
+                e.Context.HttpRequest.ResponseStatusCode == (int)ConflictStatusCode &&
+                e.Context.ReportLocation.LineNumber > 0 &&
+                !string.IsNullOrEmpty(e.Context.ReportLocation.FilePath) &&
+                e.Context.ReportLocation.FunctionName.Equals("CreateComplexContext") &&
+                e.ServiceContext.Service.Equals(ServiceName) &&
+                e.ServiceContext.Version.Equals(Version)
+            );
+        }
+
+        /// <summary>
+        /// Matcher to check if a <see cref="ReportedErrorEvent"/> matches
+        /// the <see cref="SimpleContext"/>
+        /// </summary>
+        private ReportedErrorEvent IsSimpleContext()
+        {
+            return Match.Create<ReportedErrorEvent>(e =>
+                e.Message.Contains(SimpleException.Message) &&
+                string.IsNullOrEmpty(e.Context.HttpRequest.Method) &&
+                string.IsNullOrEmpty(e.Context.HttpRequest.Url) &&
+                string.IsNullOrEmpty(e.Context.HttpRequest.UserAgent) &&
+                e.Context.HttpRequest.ResponseStatusCode == 0 &&
+                e.Context.ReportLocation.LineNumber == 0 &&
+                string.IsNullOrEmpty(e.Context.ReportLocation.FilePath) &&
+                string.IsNullOrEmpty(e.Context.ReportLocation.FunctionName) &&
+                e.ServiceContext.Service.Equals(ServiceName) &&
+                e.ServiceContext.Version.Equals(Version)
+            );
+        }
+
+        private Mock<ReportErrorsServiceClient> GetClientMock()
+        {
+            return new Mock<ReportErrorsServiceClient>();
+        }
+
+        private CloudErrorReportingExceptionLogger GetLogger(ReportErrorsServiceClient client)
+        {
+            return CloudErrorReportingExceptionLogger.Create(client, ProjectId, ServiceName, Version);
+        }
+
+        [Fact]
+        public void ShouldLog()
+        {
+            Mock<ReportErrorsServiceClient> mockClient = GetClientMock();
+
+            CloudErrorReportingExceptionLogger logger = GetLogger(mockClient.Object);
+
+            Assert.True(logger.ShouldLog(SimpleContext));
+            Assert.False(logger.ShouldLog(null));
+        }
+
+        [Fact]
+        public void Log()
+        {
+            Mock<ReportErrorsServiceClient> mockClient = GetClientMock();
+            mockClient.Setup(client => client.ReportErrorEvent(It.IsAny<string>(), It.IsAny<ReportedErrorEvent>(), null));
+
+            CloudErrorReportingExceptionLogger logger = GetLogger(mockClient.Object);
+            logger.Log(CreateComplexContext());
+
+            mockClient.Verify(client => client.ReportErrorEvent(FormattedProjectId, IsComplexContext(), null), Times.Once());
```

Comment:
Can't you just move `FormattedProjectid, IsComplexContext(), null` into the `Setup` call, and call `VerifyAll` or whatever it's called?